sentry_sdk/integrations/opentelemetry/sampler.py

```diff
@@ -124,6 +138,11 @@ def should_sample(
                 # Check if there is a traces_sample_rate
                 sample_rate = client.options.get("traces_sample_rate")
 
+        # Down-sample in case of back pressure monitor says so
+        # TODO: this should only be done for transactions (aka root spans)
+        if client.monitor:
```

Comment:
move this below the `is_valid_sample_rate` block, just before the actual random call is made

Reply:
makes sense